osh/cmd_eval.py

```diff
@@ -1451,6 +1511,19 @@ def _Dispatch(self, node, cmd_st):
 
                 status = 0
 
+            elif case(command_e.Func):
+                node = cast(command.Func, UP_node)
+
+                name = lexer.TokenVal(node.name)
+                if name in self.funcs and not self.exec_opts.redefine_proc():
```

Comment:
Probably should rename this `redefine_proc_func`

The idea there was to work around the lack of namespacing in shell

If you have `lib/foo.sh` and `lib/bar.sh` with the same function, it will alert you of the name clash at startup time

BUT in interactive mode, you want to turn that off so you can redefine!